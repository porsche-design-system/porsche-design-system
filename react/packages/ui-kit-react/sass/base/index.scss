@import "font/font";
@import "icon/icon.fontface";
@import "reset/reset";
@import "icon/icon";
@import "typo/typo";
@import "utils/utils";
