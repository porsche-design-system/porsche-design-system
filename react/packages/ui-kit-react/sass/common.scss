// @important: Only variable definitions and mixins go in here
@import "./variables";
@import "../../../node_modules/@porsche/ui-kit-core/src/mixins/px-rem-converter";
@import "../../../node_modules/@porsche/ui-kit-core/src/mixins/breakpoint";
@import "../../../node_modules/@porsche/ui-kit-core/src/base/overflow/overflow.mixin";
@import "./base/typo/typo.mixin";
@import "./base/icon/icon.mixin";
