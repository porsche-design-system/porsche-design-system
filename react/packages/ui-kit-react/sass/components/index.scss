@import "atoms/index";
@import "molecules/index";
@import "organisms/index";
@import "screens/index";
@import "structures/index";
