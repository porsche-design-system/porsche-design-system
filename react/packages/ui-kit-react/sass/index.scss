// This file includes all fontfaces, globals and components needed for ui-kit-react to be displayed properly.
// Include it once in your main scss file instead of fontaces, base and components if you don't want to pick only what you need.
@import "base/index";
@import "components/index";
