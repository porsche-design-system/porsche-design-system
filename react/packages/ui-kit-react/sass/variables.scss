@import "../../../node_modules/@porsche/ui-kit-core/src/base/spacing/spacing.setup";
@import "../../../node_modules/@porsche/ui-kit-core/src/setup/global-variables";
@import "../../../node_modules/@porsche/ui-kit-core/src/setup/color-palette.setup";

// TODO: Remove doubled variables

// Globals
$global-class-prefix: "pui-";

// Colors
$black:                 $color-palette-black;
$grey-darker:           $color-palette-grey-9;
$grey-dark:             $color-palette-grey-7;
$grey:                  $color-palette-grey-5;
$grey-light:            $color-palette-grey-3;
$grey-lighter:          $color-palette-grey-1;
$white:                 $color-palette-white;

$brand-red:             $color-palette-red-2;
$brand-red-light:       $color-palette-red-1;
$brand-red-lighter:     #f2b2ba;
$brand-red-contour:     lighten($brand-red-light, 30%);

$brand-green:           #009c51;
$brand-green-light:     #b2e1ca;
$brand-green-contour:   lighten($brand-green, 30%);

$brand-blue:            $color-palette-blue-2;
$brand-blue-light:      $color-palette-blue-1;
$brand-blue-lighter:    #bad2e0;

$brand-orange:          #faaf4b;
$brand-orange-light:    #fde7c9;

$color-facebook:        #2753a7; // Source: porsche.com
$color-twitter:         #55acee; // Source: porsche.com
$color-google:          #dc4e41; // Source: porsche.com
$color-pinterest:       #cd242b; // Source: porsche.com
$color-linkedin:        #0274b3; // Source: porsche.com
$color-whatsapp:        #25d366; // Source: whatsappbrand.com
$color-wechat:          #7bb23e; // Source: brandcolors.net/b/wechat

// Global Body
$body-txt-color:        $black;

// Spaces  |  basic-spacing x1, x2, x3, x4, x5, x6, x8, x10, ...
$basic-spacing: 8px;

/* paddings of content for each breakpoint */
$padding-horizontal-xxs: 5.2%;
$padding-horizontal-xs:  5.2%;
$padding-horizontal-s:   6.25%;
$padding-horizontal-m:   6.64%;
$padding-horizontal-l:   6.5%;

// content wrapper max width
$content-wrapper-max-width: 1600px;

// Animation timings
$transition-color:          240ms;

// Breakpoints
$breakpoints: (
    "xxs": 0,
    "xs": 480px,
    "s": 768px,
    "m": 992px,
    "l": 1200px,
    "xl": 1600px
);
