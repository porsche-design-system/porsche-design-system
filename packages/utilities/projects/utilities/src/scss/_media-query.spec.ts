import * as path from 'path';
import * as fs from 'fs';
import * as sass from 'sass';

const breakpointVariables = fs.readFileSync(path.resolve('./src/scss/lib/_breakpoint.scss'), 'utf8');
const mediaQueryMixin = fs.readFileSync(path.resolve('./src/scss/_media-query.scss'), 'utf8');
const mixin = mediaQueryMixin.replace("@import 'lib/_breakpoint.scss';", breakpointVariables);

describe('pds-media-query-min()', () => {
  it.each(['xxs', 'xs', 's', 'm', 'l', 'xl', 'xxl'])('should return correct css for breakpoint: %s', (breakpoint) => {
    const result = sass.compileString(`${mixin} div {
      @include pds-media-query-min(${breakpoint}) {
        color: deeppink;
      }
    }`);
    expect(result.css).toMatchSnapshot();
  });
});

describe('pds-media-query-max()', () => {
  it.each(['xs', 's', 'm', 'l', 'xl', 'xxl'])('should return correct css for breakpoint: %s', (breakpoint) => {
    const result = sass.compileString(`${mixin} div {
      @include pds-media-query-max(${breakpoint}) {
        color: deeppink;
      }
    }`);
    expect(result.css).toMatchSnapshot();
  });
});

describe('pds-media-query-min-max()', () => {
  it.each([
    ['xxs', 'xs'],
    ['xs', 's'],
    ['s', 'm'],
    ['m', 'l'],
    ['l', 'xl'],
    ['xl', 'xxl'],
  ])('should return correct css for breakpoint range: %s - %s', (min, max) => {
    const result = sass.compileString(`${mixin} div {
      @include pds-media-query-min-max(${min}, ${max}) {
        color: deeppink;
      }
    }`);
    expect(result.css).toMatchSnapshot();
  });
});
