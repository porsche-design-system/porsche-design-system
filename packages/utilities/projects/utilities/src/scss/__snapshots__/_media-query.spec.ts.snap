// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pds-media-query-max() should return correct css for breakpoint: l 1`] = `
"@media (max-width: 1299px) {
  div {
    color: deeppink;
  }
}"
`;

exports[`pds-media-query-max() should return correct css for breakpoint: m 1`] = `
"@media (max-width: 999px) {
  div {
    color: deeppink;
  }
}"
`;

exports[`pds-media-query-max() should return correct css for breakpoint: s 1`] = `
"@media (max-width: 759px) {
  div {
    color: deeppink;
  }
}"
`;

exports[`pds-media-query-max() should return correct css for breakpoint: xl 1`] = `
"@media (max-width: 1759px) {
  div {
    color: deeppink;
  }
}"
`;

exports[`pds-media-query-max() should return correct css for breakpoint: xs 1`] = `
"@media (max-width: 479px) {
  div {
    color: deeppink;
  }
}"
`;

exports[`pds-media-query-max() should return correct css for breakpoint: xxl 1`] = `
"@media (max-width: 1919px) {
  div {
    color: deeppink;
  }
}"
`;

exports[`pds-media-query-min() should return correct css for breakpoint: l 1`] = `
"@media (min-width: 1300px) {
  div {
    color: deeppink;
  }
}"
`;

exports[`pds-media-query-min() should return correct css for breakpoint: m 1`] = `
"@media (min-width: 1000px) {
  div {
    color: deeppink;
  }
}"
`;

exports[`pds-media-query-min() should return correct css for breakpoint: s 1`] = `
"@media (min-width: 760px) {
  div {
    color: deeppink;
  }
}"
`;

exports[`pds-media-query-min() should return correct css for breakpoint: xl 1`] = `
"@media (min-width: 1760px) {
  div {
    color: deeppink;
  }
}"
`;

exports[`pds-media-query-min() should return correct css for breakpoint: xs 1`] = `
"@media (min-width: 480px) {
  div {
    color: deeppink;
  }
}"
`;

exports[`pds-media-query-min() should return correct css for breakpoint: xxl 1`] = `
"@media (min-width: 1920px) {
  div {
    color: deeppink;
  }
}"
`;

exports[`pds-media-query-min() should return correct css for breakpoint: xxs 1`] = `
"@media (min-width: 0px) {
  div {
    color: deeppink;
  }
}"
`;

exports[`pds-media-query-min-max() should return correct css for breakpoint range: l - xl 1`] = `
"@media (min-width: 1300px) and (max-width: 1759px) {
  div {
    color: deeppink;
  }
}"
`;

exports[`pds-media-query-min-max() should return correct css for breakpoint range: m - l 1`] = `
"@media (min-width: 1000px) and (max-width: 1299px) {
  div {
    color: deeppink;
  }
}"
`;

exports[`pds-media-query-min-max() should return correct css for breakpoint range: s - m 1`] = `
"@media (min-width: 760px) and (max-width: 999px) {
  div {
    color: deeppink;
  }
}"
`;

exports[`pds-media-query-min-max() should return correct css for breakpoint range: xl - xxl 1`] = `
"@media (min-width: 1760px) and (max-width: 1919px) {
  div {
    color: deeppink;
  }
}"
`;

exports[`pds-media-query-min-max() should return correct css for breakpoint range: xs - s 1`] = `
"@media (min-width: 480px) and (max-width: 759px) {
  div {
    color: deeppink;
  }
}"
`;

exports[`pds-media-query-min-max() should return correct css for breakpoint range: xxs - xs 1`] = `
"@media (min-width: 0px) and (max-width: 479px) {
  div {
    color: deeppink;
  }
}"
`;
